### PR TITLE
Bug: AUC start 0 (which also needs Feature: C0 imputation)

### LINKED ISSUE #2: Feature: Impute initial concentrations for each subject at time zero (C0)
**Issue description**
Sample records are not taken for obvious reasons exactly when the dose is provided. It is very standard to impute these values for each subject using different strategies depending on the route of administration, the time-close collected samples and the observed kinetics of the drug. 

**Notes**
1) If a drug is not IV bolus, C0=0. 
 2) If is IV bolus and the concentration decline is monoexponential it will be extrapolated using a linear regression
 3) If there are postdoses in the first five minutes, the closes to t0 can be used for C0.

### PR DESCRIPTION
**Pull request by Gero1999**
## Issue
Closes #2 #60

## Description
After running NCA, AUC calculations always consider the discrete start (start at where first concentration sample was collected). This can indeed alter slightly the real values of the AUC. This is a consequence of 2 problems:

1) Doses times are being considered at the first measure, instead of at the real dose time
2) No imputed measure at dose time is consdiered

Change description.

## Definition of Done
C0 imputation option for the user specified in Setup > Data Selection

## How to test
- [x] At "NCA">"NCA Results", when C0 is imputed it is expected that start=AFRLT-ARRLT (being AFRLT-ARRLT= 0 in DOSNO=1)
- [x] At "NCA">"NCA Results", when C0 is not imputed it is expected that start={time of 1st observation} 
- [ ] AUC values  (auclast, aucinf...) from start=AFRLT-ARRLT are calculated when C0 is imputed
- [ ] All variables that should only be based on observed data (cmax, tmax...) are not considering C0 imputations
- [ ] Test that PKNCA also deals well with predoses as duplicates

## Contributor checklist
- [ ] Code passes lintr checks
- [ ] Code passes all unit tests
- [ ] New logic covered by unit tests
- [ ] New logic is documented

## Notes to reviewer

Anything that the reviewer should know before tacking the pull request?

### CODE REVIEW

File: R/PKNCA_impute_method_additions.R [hunk 1]
```diff
@@ -0,0 +1,29 @@
+PKNCA_impute_method_start_log <- function(conc, time, start, end, ..., options = list()) {
```
Comment (Gero1999) [inline R/PKNCA_impute_method_additions.R:1]: Hello @m-kolomanski ! I was wondering, the variable names of these functions all should start with "PKNCA_impute_method" due to PKNCA package restrictions to use them in their methods. Is it ok if I just add the names of these functions to the WORDLIST file? Will that actually make the work to pass the lint and spelling tests?

`Line 1[object_name_linter] Variable and function name style should match snake_case or SNAKE_CASE.`

Comment (m-kolomanski) [reply R/PKNCA_impute_method_additions.R:1]: Hey,

WORDLIST works only for spellchecking, does not impact the code on linter. If you wish to stop linter from evaluating particular line, use `# nolint` comment, so:
```R
PKNCA_impute_method_start_log <- function(conc, time, start, end, ..., options = list()) { # nolint
```
Would you mind providing me with the relevant docs for why that is the case?  I have failed to find anything related and I am unsure what functionality we want to use exactly.

Comment (Gero1999) [reply R/PKNCA_impute_method_additions.R:1]: Hello, is for keeping up to date with the data imputation functions that the package provides and the way it handles them, you can create your own function using the same notation, check the last section of [this vignette](https://cran.r-project.org/web/packages/PKNCA/vignettes/v08-data-imputation.html)

**Comment by Gero1999** [comment]
Detected a bug in pk.nca() when using `ADOSEDUR = 0` in the dataset. I will work on it

**Comment by js3110** [comment]
@Gero1999 I am getting a new error now when I run the IV Bolus data: 
```
Warning: Values from `exclude` are not uniquely identified; output will contain list-cols.
• Use `values_fn = list` to suppress this warning.
• Use `values_fn = {summary_fun}` to summarise duplicates.
• Use the following dplyr code to identify duplicates.
  {data} |>
  dplyr::summarise(n = dplyr::n(), .by = c(STUDYID, PCSPEC, ANALYTE, USUBJID, DOSNO,
  PPTESTCD)) |>
  dplyr::filter(n > 1L)
Warning: Error in filter: ℹ In argument: `TIME >= (lambda.z.time.first + start) |
  is.na(lambda.z.time.first)`.
ℹ In group 1: `STUDYID = "XX01"`, `PCSPEC = "SERUM"`, `ANALYTE = "Analyte01"`,
  `USUBJID = 11101`, `DOSNO = 1`.
Caused by error in `lambda.z.time.first + start`:
```

**Pull request closed without merging**